src/checker/mod.rs

```diff
@@ -110,9 +112,12 @@ impl Checker for Checkers {
             collective.extend(nlprule.check(origin, chunks)?);
         }
 
-        collective.sort();
+        // Convert into a set to remove the duplicated suggestions
+        let set: HashSet<Suggestion> = HashSet::from_iter(collective.iter().cloned());
+        let mut suggestions: Vec<Suggestion<'s>> = Vec::from_iter(set);
+        suggestions.sort();
```

Comment:
Couldn't we integrate this by just making `collective` a hashset, which would avoid one extra iteration?

Reply:
Note that the `HashSet` only would cover the exact overlaps, but not i.e. multiple mistakes that are overlapping.

Reply:
Thank you for your reviews!

1. I updated the branch so that reduces one iteration by making `collective` a HashSet.

2. I'm striving to find a better solution for the case there are multiple mistakes.

    Could I make sure my understanding is that
    > multiple mistakes that are overlapping and/or with different fix suggestions.

    means `Where checkers point out are same, but the way they suggest is different`?
    
    If so, I think we could achieve this by iterating through both resulted collections and merging the overlapped ones based on the fields of `Suggestion`: `origin`, `span`, and `range`. (Here, in this case, we don't need to use HashSet.)
    I'll appreciate it if you could give me feedback on this!